### PR TITLE
Add compression to HTTP requests

### PR DESCRIPTION
**Pull request by kage1414**

### CODE REVIEW

File: client/src/components/Elements/Article.jsx [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+import {Div, A, ElementDiv, Preview, RightSideInfo, Svg} from '../StyledComponents.js';
 
 const Article = (props) => {
 
   return (
     <Div>
-      <HasChild>
-        <Svg className="icon-article" viewBox="0 0 24 24"><ArticleSVG /></Svg>
+      <ElementDiv>
+        <Svg className="icon-article" viewBox="0 0 24 24" style={{overflow: 'visible'}}><ArticleSVG /></Svg>
```
Comment (fredrosselet) [inline client/src/components/Elements/Article.jsx:11]: glad to know that this worked!

File: client/src/components/CourseContent.jsx [hunk 2]
```diff
@@ -18,8 +18,8 @@ class CourseContent extends React.Component {
       course: {},
       isLoaded: false,
       allExpanded: false,
-      host: 'ec2-18-130-234-175.eu-west-2.compute.amazonaws.com:9800',
-      // host: '127.0.0.1:9800',
+      // host: 'ec2-18-130-234-175.eu-west-2.compute.amazonaws.com:9800',
+      host: '127.0.0.1:9800',
```
Comment (fredrosselet) [inline client/src/components/CourseContent.jsx]: not sure I understand why you had to switch back to localhost. Is that temporary while working on it locally?

File: server/controller.js [hunk 3]
```diff
@@ -93,8 +93,8 @@ module.exports.bundle = (req, res, next) => {
     for (let i = 0; i < dir.length; i++) {
       let filename = dir[i];
       let elements = filename.split('.');
-      let ext = elements[elements.length - 2];
-      if (ext === 'js') {
+      let ext = elements[elements.length - 1];
+      if (ext === 'gz') {
```
Comment (fredrosselet) [inline server/controller.js:97]: is the extension only one character long?

File: server/index.js [hunk 4]
```diff
@@ -7,7 +7,9 @@ const bodyParser = require('body-parser');
 const cors = require('cors');
 const PORT = process.env.PORT || 9800;
 
-app.use(compression());
+app.use(compression({
+  level: 9
+}));
```
Comment (fredrosselet) [inline server/index.js:12]: sounds cool.

File: client/src/components/Section.jsx [hunk 5]
```diff
@@ -66,15 +68,19 @@ class Section extends React.Component {
...
-            <ElementsContainer>
-              <Ul>
-                {this.props.section.elements.map(element =>
-                  <Element element={element} key={`element${element.elementId}`} kind={element.kind} />
-                )}
-              </Ul>
-            </ElementsContainer>
-          </SectionElementsBlock>
+          {/* <React.suspense fallback={<div>Loading...</div>}> */}
+          {this.props.section.elementDisplay === 'block' &&
```
Comment (fredrosselet) [inline client/src/components/Section.jsx:69]: nice.

File: client/src/components/CourseContent.jsx (hunk 2 above)
Comment (kage1414) [reply client/src/components/CourseContent.jsx]: Whoops didn't mean to do this

**Pull request closed without merging**